sqlglot/lineage.py

```diff
@@ -276,8 +276,81 @@ def to_node(
                 reference_node_name=reference_node_name,
                 trim_selects=trim_selects,
             )
+
+        elif scope.pivots:
+            pivot = next(
+                (p for p in scope.pivots if p.alias_or_name == c.table and not p.unpivot), None
+            )
+            if pivot and isinstance(pivot.args.get("columns"), list):
+                # The source is a pivot operation, so we need to trace back to the aggregated column
+                pivot_column_mapping = {}
+                pivot_aggs_count = len(pivot.expressions)
+                columns = pivot.args["columns"]
+                columns_count = len(columns)
+
+                if c.alias_or_name not in [i.this for i in columns]:
```

Comment:
Let's compute `column_name = c.name` once, after `columns_count` (there shouldn't be an alias), and then do:

```suggestion
                if any(column_name == pivoted_column.name for pivoted_column in columns):
```

Reply:
Can use `column_name` in L329 too.

Reply:
Adapted the code to compute `column_name` once and use it down the line